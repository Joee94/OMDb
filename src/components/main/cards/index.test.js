import * as React from 'react';
import renderer from 'react-test-renderer';
import Cards from './index';

jest.mock('./card', () => () => <mock-Card />);

describe('Cards', () => {
	describe('when there are search results', () => {
		const contextValues = {
			state: {
				movies: [
					{ Title: 'star wars', Poster: 'star wars poster', Year: 1999, imdbID: 'id0' },
					{ Title: 'back 2 the future', Poster: 'b2tf poster', Year: 2000, imdbID: 'id1' }
				],
				response: { error: false }
			},
			actions: 'mock-actions'
		};
		jest.spyOn(React, 'useContext').mockImplementation(() => contextValues);
		const component = renderer.create(<Cards />);
		it('should create the Cards component', () => {
			expect(component).toMatchSnapshot();
		});
	});
	describe('when there are no search results', () => {
		const contextValues = {
			state: { movies: [], searchValue: 'hello', response: { error: true } }
		};
		jest.spyOn(React, 'useContext').mockImplementation(() => contextValues);
		const component = renderer.create(<Cards />);
		it('should return the no results message', () => {
			expect(component).toMatchSnapshot();
		});
	});
	describe('when there hasnt been a search', () => {
		const contextValues = {
			state: { movies: undefined, response: { error: true } },
			actions: 'mock-actions'
		};
		jest.spyOn(React, 'useContext').mockImplementation(() => contextValues);
		const component = renderer.create(<Cards />);
		it('should return nothing', () => {
			expect(component).toMatchSnapshot();
		});
	});
});
