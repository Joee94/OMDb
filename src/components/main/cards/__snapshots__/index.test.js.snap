// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards when there are no search results should return the no results message 1`] = `
<h2>
  No search results
</h2>
`;

exports[`Cards when there are search results should create the Cards component 1`] = `
<ul>
  <li>
    <mock-Card />
  </li>
  <li>
    <mock-Card />
  </li>
</ul>
`;

exports[`Cards when there hasnt been a search should return nothing 1`] = `null`;
